File: app/src/main/java/com/locallinkonline/stcatherineschool/room/entity/MenuEntity.java
package com.locallinkonline.stcatherineschool.room.entity;

import com.google.gson.annotations.SerializedName;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@Entity(tableName = "menu_table")
public class MenuEntity {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name ="identifier")
    @SerializedName("identifier")
    private String identifier;

    @ColumnInfo(name ="menu_item")
    @SerializedName("menuItem")
    private String menuItem;

    @ColumnInfo(name ="display_name")
    @SerializedName("displayName")
    private String displayName;
}

File: app/src/main/java/com/locallinkonline/stcatherineschool/room/entity/MenuEntity.kt
package com.locallinkonline.stcatherineschool.room.entity

import com.google.gson.annotations.SerializedName
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "menu_table")
class MenuEntity(@PrimaryKey
                 @ColumnInfo(name = "identifier")
                 @SerializedName("identifier")
                 var identifier: String,
                 @ColumnInfo(name = "menu_item")
                 @SerializedName("menuItem")
                 var menuItem: String,

                 @ColumnInfo(name = "display_name")
                 @SerializedName("displayName")
                 var displayName: String


)
